packages/mobx/__tests__/decorators_20223/stage3-decorators.ts

```diff
@@ -0,0 +1,1110 @@
+})
+
+test("inheritance", () => {
+    class A {
+        @observable accessor a = 2
+    }
+
+    class B extends A {
+        @observable accessor b = 3
+        @computed
+        get c() {
+            return this.a + this.b
+        }
+        constructor() {
+            super()
+            makeObservable(this)
+        }
+    }
+    const b1 = new B()
+    const b2 = new B()
+    const values: any[] = []
+    mobx.autorun(() => values.push(b1.c + b2.c))
+
+    b1.a = 3
+    b1.b = 4
+    b2.b = 5
+    b2.a = 6
+
+    t.deepEqual(values, [10, 11, 12, 14, 18])
+})
+
+test("inheritance overrides observable", () => {
+    class A {
+        @observable accessor a = 2
+    }
+
+    class B {
+        @observable accessor a = 5
+        @observable accessor b = 3
+        @computed
+        get c() {
+            return this.a + this.b
+        }
+    }
+
+    const b1 = new B()
+    const b2 = new B()
+    const values: any[] = []
+    mobx.autorun(() => values.push(b1.c + b2.c))
+
+    b1.a = 3
+    b1.b = 4
+    b2.b = 5
+    b2.a = 6
+
+    t.deepEqual(values, [16, 14, 15, 17, 18])
+})
+
+test("reusing initializers", () => {
+    class A {
+        @observable accessor a = 3
+        @observable accessor b = this.a + 2
+        @computed
+        get c() {
+            return this.a + this.b
+        }
+        @computed
+        get d() {
+            return this.c + 1
+        }
+    }
+
+    const a = new A()
+    const values: any[] = []
+    mobx.autorun(() => values.push(a.d))
+
+    a.a = 4
+    t.deepEqual(values, [9, 10])
+})
+
+test("enumerability", () => {
+    class A {
+        @observable accessor a = 1 // enumerable, on proto
+        @computed
+        get b() {
+            return this.a
+        } // non-enumerable, (and, ideally, on proto)
+        @action
+        m() {} // non-enumerable, on proto
+    }
+
+    const a = new A()
+
+    // not initialized yet
+    let ownProps = Object.keys(a)
+    let enumProps: string[] = []
+    for (const key in a) enumProps.push(key)
+
+    t.deepEqual(ownProps, [])
+
+    t.deepEqual(enumProps, [])
+
+    t.equal("a" in a, true)
+    // eslint-disable-next-line
+    t.equal(a.hasOwnProperty("a"), false)
+    // eslint-disable-next-line
+    t.equal(a.hasOwnProperty("b"), false)
+    // eslint-disable-next-line
+    t.equal(a.hasOwnProperty("m"), false)
+
+    t.equal(mobx.isAction(a.m), true)
+
+    // after initialization
+    a.a
+    a.b
+    a.m
+
+    ownProps = Object.keys(a)
+    enumProps = []
+    for (const key in a) enumProps.push(key)
+
+    t.deepEqual(ownProps, [])
+
+    t.deepEqual(enumProps, [])
+
+    t.equal("a" in a, true)
+    // eslint-disable-next-line
+    t.equal(a.hasOwnProperty("a"), false)
+    // eslint-disable-next-line
+    t.equal(a.hasOwnProperty("b"), false)
+    // eslint-disable-next-line
+    t.equal(a.hasOwnProperty("m"), false)
+})
+
+test("issue 285 (2022.3) (legacy/field decorator)", () => {
```

Comment:
The enumerability is an issue here since `toJS` expects fields to be enumerable. Seems like that is becoming a topic of bigger conversation - `accessor`s being non-enumerable as almost part of the standard is very different.